python/cudf/cudf/core/column/string.py

```diff
@@ -682,28 +682,20 @@ def serialize(self):
     @classmethod
     def deserialize(cls, header, frames):
         # Deserialize the mask, value, and offset frames
-        arrays = []
-
-        for each_frame in frames:
-            if hasattr(each_frame, "__cuda_array_interface__"):
-                each_frame = cuda.as_cuda_array(each_frame)
-            elif isinstance(each_frame, memoryview):
-                each_frame = np.asarray(each_frame)
-                each_frame = cudautils.to_device(each_frame)
-
-            arrays.append(libcudf.cudf.get_ctype_ptr(each_frame))
+        buffers = [Buffer(each_frame) for each_frame in frames]
+        ptrs = [ctypes.c_uint64(buf.ptr) for buf in buffers]
 
         if header["null_count"] > 0:
-            nbuf = arrays[2]
+            nbuf = ptrs[2]
         else:
             nbuf = None
 
         # Use from_offsets to get nvstring data.
         # Note: array items = [nbuf, sbuf, obuf]
         scount = header["nvstrings"]
         data = nvstrings.from_offsets(
```

Comment:
We can actually just remove this entire function and construct the string column from the buffers directly now. Will be much faster as well.

Reply:
Should we be using `build_column`? Sorry still catching up with how things got restructured?

Reply:
Yes, @shwina it seems like we should build a `build_string_column` function similar to categoricals.

@jakirkham you can see how to do it here: https://github.com/rapidsai/cudf/blob/branch-0.13/python/cudf/cudf/core/column/column.py#L1111-L1118

Reply:
@jakirkham if you want to add a `build_string_column` function that would be appreciated as well 😄

Reply:
Would it be possible to defer these to another PR? Happy to take it as an action item.

Reply:
Sure, there's a massive amount of changes coming here anyway, so no worries.

Reply:
Thanks Keith! 😄